lib_eio_linux/eio_linux.ml

```diff
@@ -1160,11 +1160,87 @@ let net = object
   method getnameinfo = Eio_unix.getnameinfo
 end
 
+let pid_to_process close pid = object
+  inherit Eio.Process.process
+  method pid = pid
+  method status = 
+    match Eio_unix.run_in_systhread @@ fun () -> let v = Unix.waitpid [] pid in close (); v with
```

Comment:
You might need to handle `EINTR` here.

Reply:
Handled in [91c7c4a](https://github.com/ocaml-multicore/eio/pull/330/commits/91c7c4a9299e8639d313de7eb590722fe2b19baf)